### ISSUE TITLE
Enforce dtype for empty()

### ISSUE THREAD

**Issue by certik** [lcompilers/lpython#1790]
LPython should give an error message:
```python
    x: i32[4, 5, 2] = empty([4, 5, 2])
    y: f64[24, 100, 2, 5] = empty([24, 100, 2, 5])
```
Since in CPython the arrays would be both float (?) arrays of the default kind. Instead, it needs to enforce something like this:
```python
    from numpy import int32, float64
    x: i32[4, 5, 2] = empty([4, 5, 2], dtype=int32)
    y: f64[24, 100, 2, 5] = empty([24, 100, 2, 5], dtype=float64)
```

**Comment by Shaikh-Ubaid** [comment]
I think we can enforce. In `CPython`, I think it is optional.

**Comment by certik** [comment]
What I mean by that is that if dtype is not specified, then it will use the same dtype as CPython (I think it's float64).

Consequently, this `x: i32[4, 5, 2] = empty([4, 5, 2])` must fail, because the RHS is f64 and the LHS is i32.

**Comment by certik** [comment]
I am assigning this to @czgdp1807, since he implemented #2118 for i8, so it just needs to be extended for all dtypes.
